appcli/variables_manager.py

```diff
@@ -95,34 +98,119 @@ def set_all_variables(self, variables: Dict):
         self.__save(variables)
 
     def __get_configuration(self) -> Dict:
-        """Get the current configuration from file
+        """Get the current configuration from file(s)
 
         Returns:
             Dict: the current configuration
+
         """
-        try:
-            raw_data = self.configuration_file.read_text(encoding="utf-8")
+        config_variables_main = self.__get_configuration_main()
+        config_variables = self.__get_configuration_extra(config_variables_main)
+        return config_variables_main | config_variables
 
-            # If the file is empty, the YAML library will load as `None`. Since
-            # we expect this function to return a valid dict, we return an
-            # empty dictionary if it's empty.
-            yaml_data = self.yaml.load(raw_data)
-            if yaml_data is None:
-                return {}
-            return yaml_data
+    def __get_configuration_main(self) -> Dict:
+        """Gets the configuration from the main `settings.yml` file.
 
+        Returns:
+            Dict: The configuration data from the main configuration file.
+        """
+        config_variables_main = dict()
+        try:
+            data_string = self.__read_configuration_source(self.configuration_file)
+            config_variables_main |= self.__convert_yaml_to_dict(
+                data_string, self.configuration_file.stem
+            )
         except Exception as ex:
             raise Exception(
-                f"Could not read configuration file at [{self.configuration_file}]"
+                f"Could not read main configuration file at [{self.configuration_file}]"
             ) from ex
+        return config_variables_main
+
+    def __get_configuration_extra(self, variables: Dict) -> Dict:
```

Comment:
Prefer if we called it _get_extra_configuration, easier to understand what it's getting straight away

Reply:
Fixed.